pkg/tools/sandbox/types.go

```diff
@@ -42,7 +43,7 @@ type SupplyNSMgrFunc func(context.Context, *registryapi.NetworkServiceEndpoint,
 type SupplyRegistryFunc func(ctx context.Context, expiryDuration time.Duration, proxyRegistryURL *url.URL, options ...grpc.DialOption) registry.Registry
 
 // SupplyRegistryProxyFunc supplies registry proxy
-type SupplyRegistryProxyFunc func(ctx context.Context, dnsResolver dnsresolve.Resolver, handlingDNSDomain string, proxyNSMgrURL *url.URL, options ...grpc.DialOption) registry.Registry
+type SupplyRegistryProxyFunc func(ctx context.Context, proxyRegistryURL *url.URL, options ...proxydns.Option) registry.Registry
```

Comment:
We can have a different implementation of the registry-proxy. So let's do not use `proxydns.Option` directly.